drivers/sensor/tdk/icp101xx/icp101xx.c

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2024 TDK Invensense
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <math.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/__assert.h>
+
+#include <zephyr/logging/log.h>
+
+#include "icp101xx.h"
+
+LOG_MODULE_REGISTER(ICP101XX, CONFIG_SENSOR_LOG_LEVEL);
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#error "ICP101XX driver enabled without any devices"
+#endif
+
+void inv_icp101xx_sleep_us(int us)
```

Comment:
we don't need to redefine k_sleep

Reply:
inv_icp101xx_sleep_us is a HAL required by the ICP101xx official driver by TDK.